DllImportGenerator/DllImportGenerator/StubCodeContext.cs

```diff
@@ -46,7 +46,14 @@ public enum Stage
             /// <summary>
             /// Keep alive any managed objects that need to stay alive across the call.
             /// </summary>
-            KeepAlive
+            KeepAlive,
+
+            /// <summary>
+            /// Convert native data to managed data
+            /// where native values will leak if we
+            /// fail to unmarshal, but do run cleanup.
+            /// </summary>
+            LeakSafeUnmarshal
```

Comment:
It may be useful to have a comment that describes the overall structure of the method to make it clear how the stages fit together. Something like: 
```
... Setup ...
try
{
    ... Marshal ...
    fixed (... Pin ...)
    {
        ... Invoke ...
    }
    ... Unmarshal ...
}
finally
{
    ... Cleanup...
}

... KeepAlive ...
```

Reply:
This is something we are attempting to push into the design documents: https://github.com/dotnet/runtimelab/blob/DllImportGenerator/DllImportGenerator/designs/Pipeline.md

Which also need to be updated with these new stages.